libs/geo/src/main/java/org/elasticsearch/geo/utils/WellKnownText.java

```diff
@@ -58,11 +57,11 @@
     private final String EOL = "END-OF-LINE";
 
     private final boolean coerce;
-    private final boolean ignoreZValue;
+    private final GeometryValidator validator;
 
-    public WellKnownText(boolean coerce, boolean ignoreZValue) {
+    public WellKnownText(boolean coerce, GeometryValidator validator) {
```

Comment:
Should coerce also be part of the validator? Or are we trying to ensure the Validator treats incoming Geometry as "immutable"? It certainly makes the API cleaner, and `XY` geometry will coerce in a slightly different manner than `LatLon` geometry.

Reply:
Hmm... or maybe coerce is specific to the builders? And `closeLinearRingIfCoerced` is called there instead?

Reply:
Yeah here coerce is basically close linear ring if they are open. I am not really sure what to do with it to be honest. I kind of hope that we can deprecate that and remove it eventually. But I am ok with allowing validator to do coercing if needed but I am not totally sold that it belongs there to be honest.

Reply:
Sounds good. I don't think it needs to hold up this PR by any means. Let's see where the full refactoring goes and the solution may become more clear down the line.